src/components/tabs/tabs.ts

```diff
@@ -384,38 +384,34 @@ export class Tabs extends Ion implements AfterViewInit {
     // it's possible the tab is only for opening modal's or signing out
     // and doesn't actually have content. In the case there's no content
     // for a tab then do nothing and leave the current view as is
-    if (!selectedTab.root) {
-      selectedTab.ionSelect.emit(selectedTab);
-      this.ionChange.emit(selectedTab);
-      return;
-    }
-
-    // At this point we are going to perform a page switch
-    // Let's fire willLeave in the current tab page
-    let currentPage: ViewController;
-    if (currentTab) {
-      currentPage = currentTab.getActive();
-      currentPage && currentPage._willLeave(false);
-    }
+    if (selectedTab.root) {
+      // At this point we are going to perform a page switch
+      // Let's fire willLeave in the current tab page
+      var currentPage: ViewController;
+      if (currentTab) {
+        currentPage = currentTab.getActive();
+        currentPage && currentPage._willLeave(false);
+      }
 
-    // Fire willEnter in the new selected tab
-    const selectedPage = selectedTab.getActive();
-    selectedPage && selectedPage._willEnter();
+      // Fire willEnter in the new selected tab
+      const selectedPage = selectedTab.getActive();
+      selectedPage && selectedPage._willEnter();
 
-    // Let's start the transition
-    opts.animate = false;
-    selectedTab.load(opts, () => {
-      if (opts.updateUrl !== false) {
-        this._linker.navChange(DIRECTION_SWITCH);
-      }
-      this._tabSwitchEnd(selectedTab, selectedPage, currentPage);
-    });
-  }
+      // Let's start the transition
+      opts.animate = false;
+      selectedTab.load(opts, () => {
+        if (opts.updateUrl !== false) {
+          this._linker.navChange(DIRECTION_SWITCH);
+        }
+        this._tabSwitchEnd(selectedTab, selectedPage, currentPage);
+      });
+    }
 
-  _tabSwitchEnd(selectedTab: Tab, selectedPage: ViewController, currentPage: ViewController) {
     selectedTab.ionSelect.emit(selectedTab);
     this.ionChange.emit(selectedTab);
```

Comment:
Since _tabSwitchEnd() is called from inside a callback, will ionChange always emit after that function has executed? It's not clear whether that'll always be the case or not. If it is, then this LGTM.